static/js/actions/index.js

```diff
@@ -137,6 +145,10 @@ export function fetchDashboard() {
     dispatch(requestDashboard());
     return api.getDashboard().
       then(dashboard => dispatch(receiveDashboardSuccess(dashboard))).
-      catch(() => dispatch(receiveDashboardFailure()));
+      catch(e => {
+        dispatch(receiveDashboardFailure());
+        // propogate exception
+        return Promise.reject(e);
+      });
```

Comment:
It depends. If a `throw` happened in either `getDashboard` or in the `then()` immediately afterward, it would have the exception. But if there's a rejected promise then `e` is just whatever the promise value happened to be. In our `fetchJSONWithCSRF` function, if we get a non-2xx status code the value would be the content of the response